### ISSUE TITLE
(guess) failed to call xQueueCreate

### ISSUE THREAD

**Issue by vkcorp** [Xinyuan-LilyGO/T-SimHat#1]
I prepraed the really sime code as below, with my 7000G and T-SimHat.
I think setup() wouldn't be finished forever, becuase of fail at calling xQueueCreate(),
so I just see the message "loop" and no "Setup.finished" on Serial monitor.

```
#include <ESP32CAN.h>
#include <CAN_config.h>

CAN_device_t CAN_cfg;

void setup() 
{
    Serial.begin(115200);

    CAN_cfg.speed = CAN_SPEED_500KBPS;
    CAN_cfg.tx_pin_id = (gpio_num_t)22;
    CAN_cfg.rx_pin_id = (gpio_num_t)21;
    CAN_cfg.rx_queue = xQueueCreate(10, sizeof(CAN_frame_t));
    
    ESP32Can.CANInit();
    Serial.println("Setup.finished");
}

void loop() 
{
    delay(1000);
    Serial.println("loop");
}
```

Is there any problem of my codes ?

Below code was also failed as the same reason of my code.
(Actullay my code was prepared to find why after running blow one)

https://github.com/Xinyuan-LilyGO/T-SimHat/blob/main/examples/simple/simple.ino

Without a line of xQueueCreate() or make it NULL, 
it could be launched with setup() but some kind of unknown error messages were full in Serial mornitor :(

**Comment by LilyGO** [comment]
Hello。Is there any error message? Is your current ESP package up to date

**Comment by vkcorp** [comment]
No error message. 
I think I can't launch my code if there is any error.

And which kind of ESP package do you mean ?
In case of below 2 packages, it's up to dated in yesterday :
- [package_esp32_index.json](https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json)
- [ESP32-Arduino-CAN](https://github.com/miwagner/ESP32-Arduino-CAN)

Actually I downloaded all things in yesterday.

**Comment by vkcorp** [comment]
Do we have to define "CAN_cfg.rx_queue" ? Is it mandatory ?
If I can avoid to use xQueueCreate with another method, I will try...

**Comment by LilyGO** [comment]
What error does this [sketch ](https://github.com/Xinyuan-LilyGO/T-SimHat/blob/main/examples/simple/simple.ino) report?

**Comment by vkcorp** [comment]
In that scketch, there was no definition for below :

#define RS485_EN_PIN 23
#define RS485_TX_PIN 19
#define RS485_RX_PIN 18
#define CAN_TX_PIN   22
#define CAN_RX_PIN   21

p.s I think I need to make another ticket for this, it will help to other perople who has same question like me :)

**Comment by vkcorp** [comment]
After fixing problem in simple.ino, I lauched it on my board.
Then I can expect below lines will be printed at my console, but nothing..

>
void CANTask(void *prarm)

    Serial.print("CAN SPEED :");
    Serial.println(CAN_cfg.speed);

**Comment by LilyGO** [comment]
Can you post a video demonstrating the bug you found? I didn't have the problem you mentioned in my test.

**Comment by vkcorp** [comment]
I think nothing special in my code, just adding 2 lines at simple.ino

- a last line of void setup() : Serial.println("setup");
- a last line of void loop() : Serial.println("loop");

- and there is already expect to appear a line at  inside CANTask(void *prarm) : Serial.println(CAN_cfg.speed);

But in my console, only "loop" is prined as below capture shot :

![image](https://user-images.githubusercontent.com/6039321/181695596-b689b2b7-839f-432f-9625-6d9df5e5c9a5.png)

I guess Serial.println() wouldn't be printed even functions are launched...

**Comment by LilyGO** [comment]
Serial.println(CAN_cfg.speed);  Print it only once. You don't need to print extra. Try resetting to see if it prints

**Comment by vkcorp** [comment]
Yes I just want to see it, "once". 
There is no time to see it in my console messages !! That's my point and I wonder why it's not appeared ?

**Comment by LilyGO** [comment]
Can you make a video demonstration?

**Comment by vkcorp** [comment]
Well, I guess there are many things to capture, what do you want to see in focus ?
Code or console message ? Or Uploading messages ?
Maybe not a boad it's just a board without any connection yet..

**Comment by LilyGO** [comment]
1. Burn the simple firmware (do not make any changes to the code) 
2.  Open the serial monitor
3. Reset the board

**Comment by vkcorp** [comment]
Here is capture file for you. Nothing changed in code!

https://drive.google.com/file/d/1onH3qqvUVSneFU4bgsHN_VbZSEUUu_Ey/view?usp=sharing

There are no messages in serial monitor!

**Comment by LilyGO** [comment]
Can we put the board in the video? Any reset board look serial port

**Comment by vkcorp** [comment]
This is my boad. It's blinking and clicker sound from delay module when I lauching the sample code.

https://drive.google.com/file/d/1Zf5zVCL8chgBGBlW8GpDZcaAHosq8uAR/view?usp=sharing

**Comment by LilyGO** [comment]
You did not reset the board to see if the serial port output is normal. A relay issuing a clicker indicates that it is running properly. 
The serial monitor of the arduino sometimes doesn't work well. You should download the useful serial port tool. If you want to verify that the can bus is normal. The best way is to have two boards use the CAN bus communication to verify the sending and receiving.

**Comment by vkcorp** [comment]
> You did not reset the board to see if the serial port output is normal. 

Could you tell me how to reset the board for the serial port ?

**Comment by LilyGO** [comment]
The RST button is available on the sim7000G board
